samples/system-test/products.test.js

```diff
@@ -208,4 +208,32 @@ describe(`products`, () => {
       assert.ok(err.message.includes('Not found'));
     }
   });
+
+  it(`should remove orphaned products`, async () => {
+    const newProductId = `ProductId${uuid.v4()}`;
+    const newProductPath = productSearch.productPath(
+      testProduct.projectId,
+      testProduct.location,
+      newProductId
+    );
+
+    assert.strictEqual(await getProductOrFalse(newProductPath), false);
+    let output = execSync(
+      `${cmd} createProduct "${testProduct.projectId}" "${testProduct.location}" "${newProductId}" "${testProduct.productDisplayName}" "${testProduct.productCategory}"`
+    );
+
+    assert.match(output, new RegExp(`Product name: ${newProductPath}`));
+
+    output = execSync(
+      `${cmd} purgeOrphanProducts "${testProduct.projectId}" "${testProduct.location}"`
+    );
+
+    assert.match(output, new RegExp(`Orphan products deleted.`));
+    try {
+      await productSearch.getProduct({name: `${newProductPath}`});
```

Comment:
I'm not a fan of try/catch blocks in tests and suspect it might be redundant - is there any way that `Orphan products deleted.` would be printed if they were in fact, not deleted? If not, then you can remove the block. If there is, then I can think harder and help you figure out a non try/catch way to check this logic.

Reply:
I was following the pattern from the existing 'delete product' test. I think it makes sense to double-check deletions and I don't seen another way to do it in this library because there isn't a 'product exists' method. So #getProduct seems the only viable option to me and that really only has a success/error result. Let me know if you can think of another approach